core-common/src/test/java/org/zalando/nakadi/annotations/validation/DataLakeAnnotationsTest.java

```diff
@@ -44,6 +44,18 @@ public void whenMaterializationEventFormatIsWrongThenFail() {
                         DataLakeAnnotationValidator.MATERIALISE_EVENTS_ANNOTATION)));
     }
 
+    @Test
+    public void whenMaterializationEventIsOnNoRetentionPeriodFail() {
+        final var annotations = Map.of(
+                DataLakeAnnotationValidator.MATERIALISE_EVENTS_ANNOTATION, "on"
+        );
+        final Set<ConstraintViolation<TestClass>> result = validator.validate(new TestClass(annotations));
+        assertTrue("Annotation " + DataLakeAnnotationValidator.RETENTION_REASON_ANNOTATION + " is required," +
```

Comment:
The "reason" annotation seems unrelated here, or am I missing something?

Reply:
> The "reason" annotation seems unrelated here, or am I missing something?

@adrianafirutanasie can you check that, please?

Reply:
the validation is that when MATERIALISE_EVENTS_ANNOTATION=on, the RETENTION_REASON_ANNOTATION should also be present.